generator/src/test/java/org/stjs/generator/writer/fields/FieldsGeneratorTest.java

```diff
@@ -258,9 +260,28 @@ public void testTwoDimensArrayLoop() {
 				"            sum += element[0];\n" +
 				"        }");
 	}
+	
 
 	@Test
 	public void testProtectedMethodInParent() {
 		assertCodeContains(Fields31_protected_parent.class, "return this._parent._getProtectedField();");
+	}	
+
+	@Test
+	public void testCreatedFixedArraySize() {
+		assertCodeContains(Fields31_create_fixed_array_size.class, "" +
+				"    this._aStringArray = newJavaArray(10);\n" +
+				"    this._aStringArray = newJavaArray(this.constructor.getSize2());\n" +
+				"    this._aDoubleStringArray = newJavaArray(200, 300);\n" +
+				"    this._aDoubleStringArray = newJavaArray(this.constructor.getSize2(), 300);\n" +
+				"    this._aDoubleStringArray = newJavaArray(this.constructor.getSize2(), this.constructor.getSize3());\n");
 	}
+
+	@Test
+	public void testInitializeFixedArray2Dimensions() {
+		String executeResult = (String) execute(Fields32_array2Dimensions.class);
+
+		Assert.assertEquals("[null,null,null][null,null,null]", executeResult);

```

Comment:
I believe there is a bug somewhere, a 2x3 dimensions array results should be `[[null, null, null], [null, null, null]]`.